core/lib/multivm/src/versions/vm_latest/types/internals/transaction_data.rs

```diff
@@ -212,16 +211,16 @@ impl TransactionData {
             self.reserved_dynamic.len() as u64,
         );
 
-        let coefficients = OverheadCoefficients::from_tx_type(self.tx_type);
-        get_amortized_overhead(
-            total_gas_limit,
-            gas_price_per_pubdata,
-            encoded_len,
-            coefficients,
-        )
+        derive_overhead(encoded_len)
     }
 
-    pub(crate) fn trusted_ergs_limit(&self, _block_gas_price_per_pubdata: u64) -> U256 {
+    pub(crate) fn trusted_ergs_limit(&self) -> U256 {
+        if self.tx_type == L1_TX_TYPE {
+            // In case we get a users' transactions with unexpected gas limit, we do not let it have more than
+            // a certain limit
+            return U256::from(PRIORITY_TX_MAX_GAS_LIMIT);
```

Comment:
should it be `min(self.gas_limit, PRIORITY_TX_MAX_GAS_LIMIT)`?

Reply:
Not necessarily (the bootloader would never give more gas than the transaction offers), but yes, better to add the `min` for more clarity